scripts/ts/src/update-metadata.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  findMetadataPda,
+  safeFetchMetadata,
+  updateV1,
+} from "@metaplex-foundation/mpl-token-metadata";
+import {
+  keypairIdentity,
+  publicKey as umiPublicKey,
+} from "@metaplex-foundation/umi";
+import { base58 } from "@metaplex-foundation/umi/serializers";
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import { fromWeb3JsKeypair } from "@metaplex-foundation/umi-web3js-adapters";
+import { PublicKey } from "@solana/web3.js";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+import { anchorOptions, createAnchorProvider } from "./anchor-options.js";
+
+export const main = async (argv: string[] = hideBin(process.argv)) => {
+  const args = await yargs(argv)
+    .command(
+      "* <mint>",
+      "Update the Metaplex metadata for the provided mint address",
+    )
+    .options({
+      ...anchorOptions,
+      name: {
+        type: "string",
+        description: "Token name",
+      },
+      symbol: {
+        type: "string",
+        description: "Token symbol",
+      },
+      tokenUri: {
+        type: "string",
+        description: "Token metadata uri",
+      },
+    })
+    .positional("mint", {
+      type: "string",
+      description: "Mint to set the metadata for",
+      demandOption: true,
+      coerce: (mint: string) => new PublicKey(mint),
+    })
+    .parse();
+
+  const provider = createAnchorProvider(args);
+
+  const umi = createUmi(provider.connection.rpcEndpoint);
+  if (provider.wallet.payer === undefined) {
```

Comment:
this won't work with ledger unlike the other scripts here, maybe leave a comment explaining that